elements/pfelement/src/pfelement.js

```diff
@@ -398,6 +387,20 @@ class PFElement extends HTMLElement {
     this.log(`render`);
     this.resetContext();
 
+    // If the slot definition exists, set up an observer
+    if (typeof this.slots === "object" && this._slotsObserver) {
+      this._slotsObserver.observe(this, { childList: true });
+      this._initializeSlots(this.tag, this.slots);
+    }
+
+    // If an observer was defined, set it to begin observing here
+    if (this._cascadeObserver)
```

Comment:
@kylebuch8 Same question around the cascade observer.  Maybe it makes sense to wait to attach the cascade observer until the component upgrades?

Reply:
I'm thinking about scenarios where the element is re-running the render() method when properties change.  Would it be duplicating the observer?

Reply:
Shoot yeah I think you're right.  What's a better place to include it or is there a way to query for it's existence first?

Reply:
So this is interesting.  The `observe` method is smart enough to remove and add and remove observers on its own.  

> If you call observe() on a node that's already being observed by the same MutationObserver, all existing observers are automatically removed from all targets being observed before the new observer is activated.

https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#reusing_mutationobservers

Reply:
Nice! Awesome find. I was seeing these observers from pfelement repeatedly run in content set during the updating of the rendering.

Reply:
Did some superficial performance testing and didn't see any glaring differences between moving this into the render method and keeping it in connectedCallback.   I wonder if we could get @zhawkins on the case.